input/posts/2020-04-17-i-ported-my-blog-to-statiq.md

```diff
@@ -0,0 +1,368 @@
+
+<br/>
+
+### Attempt 1 - Porting Wyam Blog recipe to Statiq
+My first though was to port the old Wyam blog recipe and theme to Statiq. So, I cloned Wyam sources and started porting these to Statiq.
+* [Wyam Blog Recipe](https://github.com/Wyamio/Wyam/tree/94a3f1ba258b7d1aaf4f9e55b222697698346396/src/recipes/Wyam.Blog)
+* [Phantom Blog Theme](https://github.com/Wyamio/Wyam/tree/94a3f1ba258b7d1aaf4f9e55b222697698346396/themes/Blog/Phantom)
+
+After a couple of hours working on this, which can be found [here](https://github.com/mholo65/mholo65/commits/feature/statiq), I quickly realized that this is very difficult and time consuming. And worst of all, I was porting code which I had no idea of what it was doing. The Wyam blog recipe was very extensible and I did not need many of the extension points it was offering. Some of the code would also have been better off just re-written in Statiq. Once again, I consulted Dave...
+
+### Attempt 2 - Using Statiq.Web
+Once Dave was happy with [Statiq Framework](https://github.com/statiqdev/Statiq.Framework), which is the core of Statiq, he started working on the replacements for Wyam recipes, namely [Statiq.Web](https://github.com/statiqdev/Statiq.Web) and [Statiq.Docs](https://github.com/statiqdev/Statiq.Docs). So in beginning of March I decided to give Statiq.Web a go (source code [here](https://github.com/mholo65/mholo65/commits/statiq.web)). I quickly faced same problem here as with [Attempt 1](#attempt-1-porting-wyam-blog-recipe-to-statiq). While the recipe was mostly ported, it still lacked themes, which led me into porting the old Wyam themes to Statiq. The result would have been much like using Wyam, i.e. magic which I did not understand :) So I wanted full control...
+
+### Attempt 3 - Using Statiq.Framework
+As I had used [Statiq Framework](https://github.com/statiqdev/Statiq.Framework) previously for work related stuff, I was quite comfortable with this approach. I started by [contributing a Handlebars module to Statiq](https://github.com/statiqdev/Statiq.Framework/pull/90), because I prefer using Handlebars over Razor for simple templates. Once the Handlebars module was pulled in, I started porting my blog. The first thing I did was to take the HTML output from the old Wyam-powered blog and create handlebars templates from it. next step was to create some pipelines. The basics of the pipelines are described in the following sections. To better understand the concept of Statiq pipelines and phases, please read [Statiq's offical documentation](https://statiq.dev/framework/concepts/pipelines). The full source for the pipelines that creates my blog can be found on [GitHub](https://github.com/mholo65/mholo65/tree/master/src/Pipelines).
```

Comment:
```suggestion
As I had used [Statiq Framework](https://github.com/statiqdev/Statiq.Framework) previously for work related stuff, I was quite comfortable with this approach. I started by [contributing a Handlebars module to Statiq](https://github.com/statiqdev/Statiq.Framework/pull/90), because I prefer using Handlebars over Razor for simple templates. Once the Handlebars module was pulled in, I started porting my blog. The first thing I did was to take the HTML output from the old Wyam-powered blog and create handlebars templates from it. The next step was to create some pipelines. The basics of the pipelines are described in the following sections. To better understand the concept of Statiq pipelines and phases, please read [Statiq's offical documentation](https://statiq.dev/framework/concepts/pipelines). The full source for the pipelines that create my blog can be found on [GitHub](https://github.com/mholo65/mholo65/tree/master/src/Pipelines).
```